### ISSUE TITLE
Support NATS backend via websocket protocol

### ISSUE THREAD

**Issue by lynic** [k3s-io/kine#372]
Besides nats:// protocol, NATS server also support ws:// or wss:// protocol. 

I'm wondering is there any plan to support nats websocket protocol?

https://github.com/k3s-io/kine/blob/59c88f9b06d47d7f9338dcab688ec03f1ef751fd/pkg/drivers/nats/config.go#L160

**Comment by brandond** [comment]
Is there a difference, or is it just a different uri scheme?

**Comment by lynic** [comment]
It's just another uri scheme, just replace "nats://" with "ws://" or "wss://" in client side.
https://docs.nats.io/running-a-nats-service/configuration/websocket

**Comment by brandond** [comment]
ok but it could be sql or etcd over websockets as well then - I don't think ws/wss is sufficient to uniquely identify it as nats.

**Comment by lynic** [comment]
Yes, ws/wss is not unique. Is it possible to use "nats-ws://" and "nats-wss://" for alias?

**Comment by brandond** [comment]
But why?

**Comment by lynic** [comment]
The background is that I setup a k3s with traefik, and installed k3s server on it, but I didn't expose NATS 4222 port, but instead expose nats-server websocket service via traefik ingress, it requires the client to use ws:// or wss:// to connect.

If I want to setup another k3s service and want to use NATS server as backend, it needs to point the db endpoint to ws:// or wss:// .

**Comment by brandond** [comment]
Why does it require that? The server has no idea what scheme you used to connect to it.

**Comment by lynic** [comment]
The server needs to understand the schema, it supports "nats://" , "ws://" and "wss://", to setup the nats server , needs to specific enable websocket support.
And the client side needs to use "ws://" or "wss://" protocol instead of "nats://".
Hence if I want to use this nats-server as database, the kine needs to set scheme to "ws://" or "wss://" instead of "nats://" .
